app/models/user.rb

```diff
@@ -0,0 +1,39 @@
+class User < ApplicationRecord
+  has_many :posts
+  attr_accessor :remember_token
+  before_save { self.email = email.downcase }
+
+  validates :name, presence: true, length: { maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255},
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6}, allow_nil: true
```

Comment:
- [ ] We should be using a before_create callback in the user model, check the requirements in the odin project [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_create_requirement.png)